lib/beaker/testmode_switcher/runner_base.rb

```diff
@@ -0,0 +1,27 @@
+module Beaker
+  module TestmodeSwitcher
+    # Contains functions used in both local runner and beaker runners
+    module RunnerBase
+      def get_acceptable_puppet_run_exit_codes(opts = {})
+        # Ensure only one option given
+        if [opts[:catch_changes], opts[:catch_failures], opts[:expect_failures], opts[:expect_changes]].compact.length > 1
+          raise(ArgumentError,
+                'Cannot specify more than one of `catch_failures`, \
+                 `catch_changes`, `expect_failures`, or `expect_changes` \
+                  for a single manifest')
+        end
+
+        # Return appropriate exit code
+        return [0]        if opts[:catch_changes]
+        return [0, 2]     if opts[:catch_failures]
+        return [1, 4, 6]  if opts[:expect_failures]
+        return [2]        if opts[:expect_changes]
+
+        # If no option supplied, return all exit codes, as an array,
+        # as acceptable so beaker returns a detailed output
+        (0...256)
+      end
+    end
+  end
+end
+include Beaker::TestmodeSwitcher::RunnerBase
```

Comment:
This pollutes the global namespace. Since this function is not relevant to most users of tms, rather put it into a class (see https://github.com/puppetlabs/beaker-testmode_switcher/blob/c8e5a93f46740cf12d996bf6a8aaff2d07b53307/lib/beaker/testmode_switcher/beaker_runners.rb#L8 for an example).